### PR TITLE
Add Rholang language support

### PR DESCRIPTION
**Pull request by tgrospic**
This PR adds grammar for [Rholang](http://rchain-architecture.readthedocs.io/en/latest/contracts/contract-design.html#rholang-a-concurrent-language) language.

## Description
Rholang is a fully featured, general purpose, Turing-complete programming language built from the rho-calculus. It is a behaviorally typed, r-eflective, h-igher o-rder process language and the official smart contracting language of [RChain Cooperative](https://www.rchain.coop).

The growing [group of developers](https://github.com/rchain/rchain/graphs/contributors) currently working on Rholang language and many more on the entire RChain project. Major release is scheduled for December and we have excellent [tutorial](https://developer.rchain.coop/tutorial) for the first release.

Rholang example on [lightshow](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Ftgrospic%2Frholang-sublime%2Fmaster%2FSyntaxes%2FRholang.YAML-tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fbirchmd%2Frchain%2Fblob%2Fdev-birch-rev-system%2Frholang%2Fexamples%2Fauction.rho&code=).

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Arho+contract+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/rchain/rchain/tree/master/rholang/examples
    - Sample license(s):
      - https://github.com/rchain/rchain/blob/master/rholang/LICENSE
  - [x] I have included a syntax highlighting grammar.
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: samples/Rholang/hello_world_again.rho [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// https://github.com/rchain/rchain/tree/master/rholang/examples (MIT License)
+
+// A simple HelloWorld contract demonstrating a contract being rerun
```
Comment (lildude) [inline samples/Rholang/hello_world_again.rho]: Linguist requires the use of real-world actual use samples in order to train the classifier. "Hello world" is _never_ real world code 😉. Please remove this sample.

File: samples/Rholang/tut-hello-again.rho [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// https://github.com/rchain/rchain/tree/master/rholang/examples (MIT License)
+
+new helloAgain in {
```
Comment (lildude) [inline samples/Rholang/tut-hello-again.rho]: This too looks like a "hello world" example. Please remove it.

File: samples/Rholang/tut-hello.rho [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// https://github.com/rchain/rchain/tree/master/rholang/examples (MIT License)
+
+new helloWorld in {
+  contract helloWorld(name) = {
```
Comment (lildude) [inline samples/Rholang/tut-hello.rho]: This too looks like a "hello world" example. Please remove it.

**Comment by pchaigno** [comment]
Closing in favor of #4219. We'll monitor the popularity of the extension there.

If you have an improved search query compared to the one used in #4219, please post it here and I'll update #4219.

**Comment by tgrospic** [comment]
I've added real-world examples and removed toy examples.

**Pull request closed without merging**